test/models/deploy_test.rb

```diff
@@ -241,6 +255,32 @@ def deploy!
     end
   end
 
+  describe "to_csv associations with_deleted" do
+    let(:deployer) { users(:super_admin) }
+    let(:other_user) { users(:deployer_buddy) }
+    let(:prod) { stages(:test_production) }
+    let(:prod_deploy) { deploys(:succeeded_production_test) }
+    let(:job) { jobs(:succeeded_production_test) }
+
+    before { Stage.any_instance.stubs(:deploy_requires_approval?).returns true }
+
+    it "returns array with deleted object values" do
+      project.update_attributes(repository_url: "https://github.com/samson-test-org/example-project")
```

Comment:
The project fixture had an invalid repo url and was giving an invalid repo url error whenever the project was called.  Not sure if this will error in a real instance when the project and repo is deleted.  I can't reproduce it in a live account, probably because GitRepository already exists for the project.